packages/@lwc/integration-karma/test/component/dynamic-imports/index.spec.js

```diff
@@ -82,11 +86,151 @@ it('should not reuse DOM elements', () => {
+            } else {
+                expect(consoleSpy.calls.error[0][0].message).toContain(
+                    'Ignoring unknown provided slot name "bar"'
+                );
+            }
+        }
+
+        // Swap construstor and check if nodes have been reallocated.
+        elm.ctor = ContainerBar;
+
+        return Promise.resolve().then(() => {
+            expect(
+                elm.shadowRoot.querySelector('[data-id="slot-default"]').assignedSlot
+            ).toBeDefined();
+            expect(elm.shadowRoot.querySelector('[data-id="slot-bar"]').assignedSlot).toBeDefined();
+
+            if (process.env.NATIVE_SHADOW) {
+                expect(elm.shadowRoot.querySelector('[data-id="slot-foo"]').assignedSlot).toBe(
+                    null
+                );
+            } else {
+                if (process.env.NODE_ENV === 'production') {
+                    expect(consoleSpy.calls.error.length).toEqual(0);
+                } else {
+                    expect(consoleSpy.calls.error[1][0].message).toContain(
+                        'Ignoring unknown provided slot name "foo"'
+                    );
+                }
+            }
+        });
+    });
+});
+
+// Using <lwc:component lwc:is={}>
+
+it('should call the loader', () => {
+    // note, using `x-` prefix instead of `x/` because these are
+    // handled by `registerForLoad`
+    registerForLoad('x-ctor', DynamicCtor);
+    registerForLoad('x-alter', AlterCtor);
+
+    const elm = createElement('x-dynamic', { is: DynamicContainer });
+    document.body.appendChild(elm);
+
+    return Promise.resolve().then(() => {
+        const child = elm.shadowRoot.querySelector('x-ctor');
+        expect(child).toBeNull();
+        // first rendered with ctor set to undefined (nothing)
+        elm.enableCtor();
+        return waitForMacroTask(() => {
+            // second rendered with ctor set to x-ctor
+            const ctorElm = elm.shadowRoot.querySelector('x-ctor');
+            expect(ctorElm).not.toBeNull();
+            const span = ctorElm.shadowRoot.querySelector('span');
+            expect(span).not.toBeNull();
+            expect(span.textContent).toBe('ctor_html');
+            elm.enableAlter();
+            return waitForMacroTask(() => {
+                // third rendered with ctor set to x-alter
+                const alterElm = elm.shadowRoot.querySelector('x-alter');
+                expect(alterElm).not.toBeNull();
+                const span = alterElm.shadowRoot.querySelector('span');
+                expect(span).not.toBeNull();
+                expect(span.textContent).toBe('alter_html');
+                elm.disableAll();
+                return waitForMacroTask(() => {
+                    // third rendered with ctor set to null (nothing)
+                    const child = elm.shadowRoot.querySelector('x-ctor');
+                    expect(child).toBeNull();
+                });
+            });
+        });
+    });
+});
```

Comment:
FWIW, I discovered recently that `async`/`await` works just fine in our Karma tests. Dunno why I hadn't realized it before. 🤷